cvmfs/manifest.h

```diff
@@ -16,21 +16,28 @@
 namespace manifest {
 
 /**
- * The breadcrumb stores the catalog root hash and a time stamp.  It is used
- * to store the last known copy of the catalog in the cache.
+ * The breadcrumb stores the catalog root hash, its revision and a time stamp.
+ * It is used to store the last known copy of the catalog in the cache.
  */
 struct Breadcrumb {
-  Breadcrumb() : catalog_hash(), timestamp(0) {}
-  Breadcrumb(const shash::Any &h, uint64_t t) : catalog_hash(h), timestamp(t) {}
+  Breadcrumb() : catalog_hash(), timestamp(0), revision(-1ul) {}
```

Comment:
Make `-1ul` a named static constant, i.e.
```suggestion
  static const uint64_t kInvalidRevision = -1ul;
  Breadcrumb() : catalog_hash(), timestamp(0), revision(kInvalidRevision) {}
```